.github/workflows/ci-upstream.yml

```diff
@@ -0,0 +1,35 @@
+name: Upstream
+
+on:
+  schedule:
+    - cron: "0 1 * * *"
+  push:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: "ubuntu-latest"
+    if: "contains(github.event.head_commit.message, 'test-upstream') || github.event_name != 'pull_request'"
```

Comment:
I'm not 100% sure this is working as the upstream build was skipped for e5e2cd9872c3833897c8ec87bf905d2299bcd456 which includes "test-upstream" in the commit message

Reply:
It only works for the first commit in the PR. I spent some time trying to figure out if you could trigger this on the latest commit on the branch but it doesn't seem the `github.event` object has this info.